federatedscope/core/auxiliaries/model_builder.py

```diff
@@ -11,99 +11,138 @@
         f'available.')
 
 
-def get_model(model_config, local_data, backend='torch'):
+def get_shape_from_data(data, model_config, backend='torch'):
+    """
+    Extract the input shape from the given data, which can be used to build
+    the data. Users can also use `data.input_shape` to specify the shape
+    Arguments:
+        data (object): the data used for local training or evaluation
+        The expected data format:
+        1): {train/val/test: {x:ndarray, y:ndarray}}}
+        2): {train/val/test: DataLoader}
+    Returns:
+        shape (tuple): the input shape
+    """
+    # Handle some special cases
+    if model_config.type.lower() in ['vmfnet', 'hmfnet']:
+        return data['train'].n_col if model_config.type.lower(
+        ) == 'vmfnet' else data['train'].n_row
+    elif model_config.type.lower() in [
```

Comment:
For graph-level tasks, data is  `{train/val/test: DataLoader}`.

Reply:
I have fixed this issue, thx!